lib/core/test/integration/Test/Integration/Scenario/CLI/Wallets.hs

```diff
@@ -28,23 +38,160 @@ import Test.Hspec.Expectations.Lifted
+        T.unpack ep `shouldContain` cmdOk
+        _ <- expectValidJSON (Proxy @(ApiTransaction t)) op
+        cp `shouldBe` ExitSuccess
+
+        expectEventually' ctx balanceAvailable amount wDest
+        expectEventually' ctx balanceTotal amount wDest
+
+        -- delete wallet
+        Exit cd <- deleteWalletViaCLI ctx $ T.unpack (wDest ^. walletId)
+        cd `shouldBe` ExitSuccess
+
+        -- restore wallet
+        (c2, o2, e2) <- createWalletViaCLI ctx ["n"] m "\n" "secure-passphraseX"
+        c2 `shouldBe` ExitSuccess
+        T.unpack e2 `shouldContain` cmdOk
+        wRestored <- expectValidJSON (Proxy @ApiWallet) o2
+        verify wRestored
+            [ expectEventually' ctx state Ready
+            , expectCliFieldEqual walletId (wDest ^. walletId)
+            ]
+
+        -- make sure funds are there
+        Stdout o3 <- getWalletViaCLI ctx $ T.unpack (wRestored ^. walletId)
+        justRestored <- expectValidJSON (Proxy @ApiWallet) o3
+        verify justRestored
+            [ expectCliFieldEqual balanceAvailable amount
+            , expectCliFieldEqual balanceTotal amount
+            , expectCliFieldEqual walletId (wDest ^. walletId)
+            ]
+
+    it "WALLETS_CREATE_03 - Cannot create wallet that exists" $ \ctx -> do
+        Stdout m1 <- generateMnemonicsViaCLI ["--size", "24"]
+        Stdout m2 <- generateMnemonicsViaCLI ["--size", "12"]
+
+        (c1, o1, e1) <- createWalletViaCLI ctx ["n1"] m1 m2 "secure-passphrase"
+        c1 `shouldBe` ExitSuccess
+        T.unpack e1 `shouldContain` cmdOk
+        j <- expectValidJSON (Proxy @ApiWallet) o1
+        expectCliFieldEqual walletName "n1" j
+
+        (c2, o2, e2) <- createWalletViaCLI ctx ["n2"] m1 m2 "Xsecure-passphraseX"
+        c2 `shouldBe` ExitFailure 1
+        o2 `shouldBe` ""
+        T.unpack e2 `shouldContain` "This operation would yield a wallet with \
+            \the following id: " ++ T.unpack (j ^. walletId) ++ " \
+            \However, I already know of a wallet with this id."
+
+    describe "WALLETS_CREATE_04 - Wallet names" $ do
+        let walNameMax = replicate walletNameMaxLength 'ą'
+        let walNameMin = replicate walletNameMinLength 'ź'
+        let matrix =
+                [ ( show walletNameMinLength ++ " char long", walNameMin )
+                , ( show walletNameMaxLength ++ " char long", walNameMax )
+                , ( "Russian", "АаБбВвГгДдЕеЁёЖжЗз")
+                -- , ( "Polish", "aąbcćdeęfghijklłmnoóp" )
+                -- , ( "Kanji", "亜哀挨愛曖悪握圧扱宛嵐")
+                ]
+        forM_ matrix $ \(title, n) -> it title $ \ctx -> do
+            Stdout m <- generateMnemonicsViaCLI ["--size", "18"]
+            (c, o, e) <- createWalletViaCLI ctx [n] m "\n" "secure-passphrase"
+            c `shouldBe` ExitSuccess
+            T.unpack e `shouldContain` cmdOk
+            _ <- expectValidJSON (Proxy @ApiWallet) o
+            o `shouldContain` n
```

Comment:
I am not using `expectCliFieldEqual walletName (T.pack n) j` here because I was getting failures due to encoding does not match (I think)... ¯\_(ツ)_/¯

Reply:
@piotr-iohk We are probably missing:

```hs
    hSetEncoding stdout utf8
    hSetEncoding stderr utf8
```

in https://github.com/input-output-hk/cardano-wallet/blob/master/lib/http-bridge/test/integration/Main.hs#L93

So far, you probably didn't have any issue because we do setup the encoding correctly for the server, but here, since the parsing is done by the integration tests I suppose problems arise... :thinking:

Reply:
@KtorZ, I have added this as suggested, unfortunately still getting the same issues...